src/poetry/config/config.py

```diff
@@ -212,6 +212,10 @@ def _get_environment_repositories() -> dict[str, dict[str, str]]:
     def repository_cache_directory(self) -> Path:
         return Path(self.get("cache-dir")) / "cache" / "repositories"
 
+    @property
+    def artifacts_cache_directory(self) -> Path:
+        return Path(self.get("cache-dir")) / "artifacts"
```

Comment:
```suggestion
        return Path(self.get("cache-dir")).expanduser() / "artifacts"
```

Not sure if it's necessary, but it's probably safer to keep this. Can you please add this in `repository_cache_directory` as well?